ipfs swarm connect → press enter → panic

#### Version information:

go-ipfs version: 0.4.20-
Repo version: 7
System version: amd64/darwin
Golang version: go1.12.4

#### Description:

Steps to replicate:

1. Run `ipfs swarm connect`
2. Press enter

This produces the following panic:

```
$ ipfs swarm connect
ipfs: Reading from /dev/stdin; send Ctrl-d to stop.

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1a3965c]

goroutine 1 [running]:
github.com/ipfs/go-ipfs/core/commands.resolveAddresses(0x209e400, 0xc0002fc400, 0xc0002af950, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	src/github.com/ipfs/go-ipfs/core/commands/swarm.go:520 +0x48c
github.com/ipfs/go-ipfs/core/commands.peersWithAddresses(0x209e400, 0xc0002fc400, 0xc0002af950, 0x1, 0x1, 0x0, 0x0, 0xc0002fc3c0, 0xc0000cb300, 0xc00022fbe0)
	src/github.com/ipfs/go-ipfs/core/commands/swarm.go:472 +0x77
github.com/ipfs/go-ipfs/core/commands.glob..func153(0xc0002c7e30, 0x209e800, 0xc0000b0cc0, 0x207cda0, 0xc0000b0c60, 0xc00022fd28, 0x1b00b5e)
	src/github.com/ipfs/go-ipfs/core/commands/swarm.go:373 +0xbf
github.com/ipfs/go-ipfs-cmds.(*executor).Execute(0xc0000ae0f8, 0xc0002c7e30, 0x209e800, 0xc0000b0cc0, 0x207cda0, 0xc0000b0c60, 0xc0000b0cc0, 0x0)
	pkg/mod/github.com/ipfs/go-ipfs-cmds@v0.0.5/executor.go:81 +0x171
github.com/ipfs/go-ipfs-cmds/cli.Run(0x209e400, 0xc0002fc380, 0x2a5c060, 0xc0000bc000, 0x3, 0x3, 0xc00000e010, 0xc0000ae000, 0xc0000ae008, 0x1f36430, ...)
	pkg/mod/github.com/ipfs/go-ipfs-cmds@v0.0.5/cli/run.go:138 +0xa63
main.mainRet(0x0)
	src/github.com/ipfs/go-ipfs/cmd/ipfs/main.go:177 +0x421
main.main()
	src/github.com/ipfs/go-ipfs/cmd/ipfs/main.go:83 +0x22
```

--- a/core/commands/p2p.go
+++ b/core/commands/p2p.go
@@ -357,14 +357,22 @@ var p2pCloseCmd = &cmds.Command{
 
 		proto := protocol.ID(protoOpt)
 
-		listen, err := ma.NewMultiaddr(listenOpt)
-		if err != nil {
-			return err
+		var (
+			target, listen ma.Multiaddr
+		)
+
+		if l {
+			listen, err = ma.NewMultiaddr(listenOpt)
+			if err != nil {
+				return err
+			}
 		}
 
-		target, err := ma.NewMultiaddr(targetOpt)
-		if err != nil {
-			return err
+		if t {
+			target, err = ma.NewMultiaddr(targetOpt)
+			if err != nil {
+				return err
+			}
 		}
 
 		if !(closeAll || p || l || t) {

--- a/go.mod
+++ b/go.mod
@@ -97,7 +97,7 @@ require (
 	github.com/miekg/dns v1.1.8 // indirect
 	github.com/mitchellh/go-homedir v1.1.0
 	github.com/mr-tron/base58 v1.1.2
-	github.com/multiformats/go-multiaddr v0.0.3
+	github.com/multiformats/go-multiaddr v0.0.4
 	github.com/multiformats/go-multiaddr-dns v0.0.2
 	github.com/multiformats/go-multiaddr-net v0.0.1
 	github.com/multiformats/go-multibase v0.0.1

